crates/core/src/repository.rs

```diff
@@ -748,7 +756,10 @@ impl<P: ProgressBars, S> Repository<P, S> {
     ///
     /// * If the command could not be parsed.
     /// * If the thread pool could not be created.
-    pub fn warm_up_wait(&self, packs: impl ExactSizeIterator<Item = PackId>) -> RusticResult<()> {
+    pub fn warm_up_wait(
+        &self,
+        packs: impl ExactSizeIterator<Item = PackId> + Clone,
```

Comment:
Why do we require `Clone` here? This would be an API breaking change 🤔

Reply:
We need to iterate twice on the list, the first time to warm up, the second time to wait on each element.

If we want to avoid this change, we can collect the list, but that would introduce an allocation.

I can change to a collect and remove the `Clone` requirement if we want to keep the API stable.

Reply:
Actually I would prefer to leave it and merge this PR with the next breaking release. I doubt that the current public API of `warm_up` is externally used anyway..

Reply:
Yeah, sure. It feels a bit like premature optimization to me, though, not sure if it's needed at this point. If it's not really needed and there is a way around (just allocate) I don't think it is necessary to introduce a bound here at this stage - if we need to get rid of it, for any reason, this is another API breaking change. So I would be vary now.

Reply:
I propose to change the visibility to `pub(crate)` - this is a breaking change but then future changes won't be breaking anymore.

Reply:
> I propose to change the visibility to `pub(crate)` - this is a breaking change but then future changes won't be breaking anymore.

That would be also a solution, we need to check, if that would also change the Result type then to a local error. I didn't look into that, can do later.

Reply:
Looked through `warm_up_wait`, we **wouldn't** need to change the Return value to a local one. So we can just make it `pub(crate)` as proposed.

Reply:
I updated the warm_up_wait command to be crate local (and rebased)